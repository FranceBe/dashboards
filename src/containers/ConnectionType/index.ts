import ConnectionTypeContainer, {
  ConnectionType,
} from 'containers/ConnectionType/ConnectionType.container'

export { ConnectionType }

export default ConnectionTypeContainer
