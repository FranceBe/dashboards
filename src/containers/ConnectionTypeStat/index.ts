import ConnectionTypeStatContainer, {
  ConnectionTypeStat,
} from 'containers/ConnectionTypeStat/ConnectionTypeStat.container'

export { ConnectionTypeStat }

export default ConnectionTypeStatContainer
