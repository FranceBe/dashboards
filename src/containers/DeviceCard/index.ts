import DeviceCardComponent, { DeviceCard } from 'containers/DeviceCard/DeviceCard.container'

export { DeviceCard }

export default DeviceCardComponent
