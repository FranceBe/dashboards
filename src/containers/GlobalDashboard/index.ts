import GlobalDashboardContainer, {
  GlobalDashboard,
} from 'containers/GlobalDashboard/GlobalDashboard.container'

export { GlobalDashboard }

export default GlobalDashboardContainer
