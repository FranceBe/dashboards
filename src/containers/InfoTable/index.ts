import InfoTableComponent, { InfoTable } from 'containers/InfoTable/InfoTable.container'

export { InfoTable }

export default InfoTableComponent
