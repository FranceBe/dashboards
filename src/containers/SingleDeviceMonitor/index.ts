import SingleDeviceMonitorContainer, {
  SingleDeviceMonitor,
} from 'containers/SingleDeviceMonitor/SingleDeviceMonitor.container'

export { SingleDeviceMonitor }

export default SingleDeviceMonitorContainer
