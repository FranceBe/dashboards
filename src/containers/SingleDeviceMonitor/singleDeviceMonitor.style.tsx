import styled from 'styled-components'
import { spaces } from 'styles/variables'

export const LinkContainer = styled.div`
  display: flex;
  align-items: center;
`

export const CardContainer = styled.div`
  margin-top: ${spaces.large};
  padding-right: 20px;
  h2 {
    text-align: center;
  }
  .live-voltage {
    margin-top: ${spaces.large};
  }
`
