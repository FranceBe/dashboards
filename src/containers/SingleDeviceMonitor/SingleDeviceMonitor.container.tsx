// Display all information for a specific device
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft'
import Loading from 'components/Loading'
import DeviceCard from 'containers/DeviceCard'
import LiveVoltage from 'containers/LiveVoltage'
import {
  CardContainer,
  LinkContainer,
} from 'containers/SingleDeviceMonitor/singleDeviceMonitor.style'
import React from 'react'
import { RouteComponentProps } from 'react-router-dom'
import { Link } from 'react-router-dom'
import { DashboardDeviceable } from 'types/dashboardDevice'
import { momentFormatter } from 'utils/momentFormatter'
import { useFetch } from 'utils/useFetch'

export const SingleDeviceMonitor: React.FC<RouteComponentProps> = ({ match }) => {
  const { id } = match.params as { id: number }
  // Using useFetch custom hook to call Api with a specific id using current URL to get a specific device
  const { data } = useFetch<DashboardDeviceable>(`api/device/${id}`)

  if (!data || Array.isArray(data)) {
    return <Loading />
  }

  return (
    <div>
      <Link to={'/'}>
        <LinkContainer>
          <ChevronLeftIcon /> Back to dashboards
        </LinkContainer>
      </Link>
      <CardContainer>
        <h2>Monitoring - {data.serial_number}</h2>
        <DeviceCard device={data} />
        <LiveVoltage
          deviceId={id}
          initialVoltage={{
            time: momentFormatter(data.last_seen_at),
            voltage: data.voltage,
          }}
        />
      </CardContainer>
    </div>
  )
}

export default SingleDeviceMonitor
