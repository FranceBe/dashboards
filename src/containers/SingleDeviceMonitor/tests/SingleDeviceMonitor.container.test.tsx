import '@testing-library/jest-dom/extend-expect'

import { render, screen } from '@testing-library/react'
import { SingleDeviceMonitor } from 'containers/SingleDeviceMonitor'
import { createLocation, createMemoryHistory } from 'history'
import React from 'react'
import { match } from 'react-router'
import { BrowserRouter as Router } from 'react-router-dom'
import * as customHook from 'utils/useFetch'

describe('SingleDeviceMonitor', () => {
  // Set up
  // Mock react-router props
  const history = createMemoryHistory()
  const path = `/`
  const match: match<{ id: string }> = {
    isExact: false,
    params: { id: '1' },
    path,
    url: path.replace(':id', '1'),
  }
  const location = createLocation(match.url)
  const routerPropsMocked = {
    history,
    location,
    match,
  }

  // Mock fetch response so we do not need to call real API for testing
  const spy = jest.spyOn(customHook, 'useFetch')
  spy.mockReturnValue({
    data: {
      connection_type: 'ethernet',
      last_seen_at: '2021-09-26T15:10:44.987',
      mac_wifi: 'c5:d7:14:84:f8:cf',
      serial_number: 'device_0',
      sim_id: 98962001,
      status: 'connected',
      url: 'http://url.com',
      voltage: 1234.933884,
    },
    status: 'fetched',
  })

  // clean all mocks after all
  afterAll(() => {
    jest.clearAllMocks()
  })

  // SetupTest for rendering component
  const setupTest = (props = {}) => {
    return render(
      <Router>
        <SingleDeviceMonitor {...routerPropsMocked} {...props} />
      </Router>,
    )
  }

  it('should match snapshot', () => {
    const { container } = setupTest()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render "back to dashboards" link', () => {
    setupTest()

    expect(screen.getByText('Back to dashboards')).toBeInTheDocument()
  })
  setupTest()
  it('should display title', () => {
    setupTest()

    expect(screen.getByText('Monitoring - device_0')).toBeInTheDocument()
  })

  it('should display DeviceCard', () => {
    setupTest()

    expect(screen.getByTestId('device-card')).toBeInTheDocument()
  })
})
