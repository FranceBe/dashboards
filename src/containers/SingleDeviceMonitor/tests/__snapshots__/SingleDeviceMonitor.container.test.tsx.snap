// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleDeviceMonitor should match snapshot 1`] = `
<div>
  <a
    href="/"
  >
    <div
      class="sc-jrsJWt hDLwjI"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
        />
      </svg>
       Back to dashboards
    </div>
  </a>
  <div
    class="sc-kEqXSa eVZGgl"
  >
    <h2>
      Monitoring - 
      device_0
    </h2>
    <div
      class="sc-gtsrHT kVqHV"
      data-testid="device-card"
    >
      <div
        class="sc-hKFxyN dqUOqf"
      >
        <div
          class="sc-jSFjdj edKxXR"
        >
          <div
            class="sc-gKAaRy jBxjRv"
          >
            Serial Number :
          </div>
          <div
            class="sc-iCoGMd SYGid"
          >
            device_0
          </div>
        </div>
        <div
          class="sc-jSFjdj edKxXR"
        >
          <div
            class="sc-bdnxRM hNjTSz"
            data-testid="connection_type-container"
            type="ethernet"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              data-testid="icon-ethernet"
              focusable="false"
              title="ethernet"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.77 6.76L6.23 5.48.82 12l5.41 6.52 1.54-1.28L3.42 12l4.35-5.24zM7 13h2v-2H7v2zm10-2h-2v2h2v-2zm-6 2h2v-2h-2v2zm6.77-7.52l-1.54 1.28L20.58 12l-4.35 5.24 1.54 1.28L23.18 12l-5.41-6.52z"
              />
            </svg>
          </div>
          <div
            class="sc-gKAaRy jBxjRv"
          >
            Connection Type :
          </div>
          <div
            class="sc-iCoGMd SYGid"
          >
            ethernet
          </div>
        </div>
        <div
          class="sc-fujyAs eKHIgW"
          data-testid="status-container"
          status="connected"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            data-testid="signal-on"
            focusable="false"
            title="connected"
            viewBox="0 0 24 24"
          >
            <path
              d="M2 22h20V2z"
            />
          </svg>
        </div>
      </div>
      <div
        class="sc-dlnjwi gjszPZ"
      />
      <div
        class="sc-hKFxyN dqUOqf"
      >
        <div
          class="sc-jSFjdj edKxXR"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root small-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 2v11h3v9l7-12h-4l4-8z"
            />
          </svg>
          <div
            class="sc-gKAaRy jBxjRv"
          >
            Voltage :
          </div>
          <div
            class="sc-iCoGMd SYGid"
          >
            1234.933884
          </div>
        </div>
        <div
          class="sc-jSFjdj edKxXR"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root small-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
            />
          </svg>
          <div
            class="sc-gKAaRy jBxjRv"
          >
            Last seen at :
          </div>
          <div
            class="sc-iCoGMd SYGid"
          >
            September 26th 2021, 3:10:44 pm
          </div>
        </div>
      </div>
      <div
        class="sc-hKFxyN dqUOqf"
      >
        <div
          class="sc-jSFjdj edKxXR"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root small-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM8.46 14.45l-1.36-.62c.28-.61.41-1.24.4-1.86-.01-.63-.14-1.24-.4-1.8l1.36-.63c.35.75.53 1.56.54 2.4.01.86-.17 1.7-.54 2.51zm3.07 1.56l-1.3-.74c.52-.92.78-1.98.78-3.15 0-1.19-.27-2.33-.8-3.4l1.34-.67c.64 1.28.96 2.65.96 4.07 0 1.43-.33 2.74-.98 3.89zm3.14 1.32l-1.35-.66c.78-1.6 1.18-3.18 1.18-4.69 0-1.51-.4-3.07-1.18-4.64l1.34-.67c.9 1.78 1.34 3.56 1.34 5.31 0 1.74-.44 3.54-1.33 5.35z"
            />
          </svg>
          <div
            class="sc-gKAaRy jBxjRv"
          >
            Mac wifi :
          </div>
          <div
            class="sc-iCoGMd SYGid"
          >
            c5:d7:14:84:f8:cf
          </div>
        </div>
        <div
          class="sc-jSFjdj edKxXR"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root small-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 10V6c0-1.11-.9-2-2-2H4c-1.1 0-1.99.89-1.99 2v4c1.1 0 1.99.9 1.99 2s-.89 2-2 2v4c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2v-4c-1.1 0-2-.9-2-2s.9-2 2-2zm-9 7.5h-2v-2h2v2zm0-4.5h-2v-2h2v2zm0-4.5h-2v-2h2v2z"
            />
          </svg>
          <div
            class="sc-gKAaRy jBxjRv"
          >
            Sim id :
          </div>
          <div
            class="sc-iCoGMd SYGid"
          >
            98962001
          </div>
        </div>
      </div>
      <div
        class="sc-hKFxyN dqUOqf"
      >
        <div
          class="sc-jSFjdj edKxXR"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root small-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
            />
          </svg>
          <div
            class="sc-gKAaRy jBxjRv"
          >
            Url :
          </div>
          <div
            class="sc-iCoGMd SYGid"
          >
            <a
              href="http://url.com"
            >
              http://url.com
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
